### DISCUSSION TITLE
Buttons are not correctly style

### DISCUSSION THREAD

**Discussion by vmasanas** [DNNCommunity/Dnn.CommunityForums#1442] [Q&A]
Some buttons are not correctly styled when using a bootstrap theme.
Notice a button with btn-primary class at top left (added on a different module), and the blue buttons the forum module uses.
I guess those are dnnPrimaryButton, but not sure if you should use the standard bootstrap classes here instead.

![image](https://github.com/DNNCommunity/Dnn.CommunityForums/assets/4357679/13ecf785-d42b-45c3-9593-502077307d6d)

**Comment by johnhenley** [comment]
@vmasanas which bootstrap version are you using on the site? And are you using the new community-bootstrap theme in forums? Beyond that, I defer to @Timo-Breumelhof for any css/ style issues:)

**Comment by vmasanas** [comment]
I'm using the community-bootstrap forums theme and I just installed this DNN boostrap skin https://github.com/2sic/dnn-theme-bootstrap4-instant

**Comment by johnhenley** [comment]
> I'm using the community-bootstrap forums theme and I just installed this DNN boostrap skin https://github.com/2sic/dnn-theme-bootstrap4-instant
> 
> 

Ok thanks.

**Comment by johnhenley** [comment]
BTW it doesn't look like the fontawesome icons are rendering in your screenshots ? Do you see anything in the browser console for that?

**Comment by vmasanas** [comment]
Does this help?

![image](https://github.com/DNNCommunity/Dnn.CommunityForums/assets/4357679/2edd5d01-c51c-46ee-8436-1fbed4f9cb2a)

**Comment by johnhenley** [comment]
> Does this help?
> 
> 
> 
> ![image](https://github.com/DNNCommunity/Dnn.CommunityForums/assets/4357679/2edd5d01-c51c-46ee-8436-1fbed4f9cb2a)
> 
> 

@Timo-Breumelhof do you remember this #599

**Comment by Timo-Breumelhof** [comment]
@vmasanas @johnhenley 

The reason you see this is is tow fold.

A. The module does not allow us to set the css classes of those buttons. It just renders DNN buttons as that is the DNN standard. FYI, this will be the case with many other modules as well.
See the template:

```
                        <div class="dcf-split-buttons-wrap">[SPLITBUTTONS]</div>
                        <div class="dcf-button-reply">[ADDREPLY]</div>
```

B. The Bootstrap Theme does not overrule the DNN buttons with the Bootstrap Button Style.
My Themes do, which is why I never see this.

![image](https://github.com/DNNCommunity/Dnn.CommunityForums/assets/8999382/70a8d91c-5bab-4ecd-a25a-6eabe1f26a57)

For anyone wanting to add this to their Theme, see: https://github.com/DNNCommunity/DNNCommunityTheme/blob/develop/_src/theme/scss/parts/_buttons.scss

For community Forums 9 we are going to target "everything is a template" am IMO we should also do this for the buttons.

Something like

[ADDREPLY:class="btn btn-primary"]

Or something better ;-)

**Comment by Timo-Breumelhof** [comment]
@vmasanas again thank you for logging his.
 I'm closing this one as we cannot reliably fix this right now (IMO it's kind of a Theme issue)
 But I did created #881 for version 9 of the module so we can set Button classes from the Template in the Future.
